bin/spice/cmd/login.go

```diff
@@ -412,13 +417,33 @@ spice login databricks --token <access-token> --google-service-account-path /pat
 			os.Exit(1)
 		}
 
-		if db_token == "" {
-			cmd.Println("No Databricks Access Token provided, use --token")
+		db_client_id, err := cmd.Flags().GetString(clientId)
+		if err != nil {
+			slog.Error("getting flag", "flag", clientId, "error", err)
+			os.Exit(1)
+		}
+
+		db_client_secret, err := cmd.Flags().GetString(clientSecret)
+		if err != nil {
+			slog.Error("getting flag", "flag", clientSecret, "error", err)
 			os.Exit(1)
 		}
 
-		params := map[string]string{
-			api.AUTH_PARAM_TOKEN: db_token,
+		if (db_token == "" && (db_client_id == "" || db_client_secret == "")) || (db_token != "" && (db_client_id != "" || db_client_secret != "")) {
```

Comment:
Nit: possibly simpler logic

```suggestion
		if (db_token == "") == (db_client_id == "" || db_client_secret == "") {
```